terraform/eks/eks.tf

```diff
@@ -0,0 +1,235 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+data "aws_availability_zones" "available" {}
+
+locals {
+  cluster_name = "airflow"
+}
+
+module "eks" {
+  source  = "terraform-aws-modules/eks/aws"
+  version = "20.24.0"
+
+  cluster_name    = local.cluster_name
+  cluster_version = "1.30"
+
+  vpc_id     = module.vpc.vpc_id
+  subnet_ids = module.vpc.public_subnets
+
+  cluster_endpoint_public_access = true
+
+  eks_managed_node_group_defaults = {
+    ami_type                   = "AL2_ARM_64"
+    use_custom_launch_template = false
+    disk_size                  = 50
+    desired_size               = 0
+  }
+
+  eks_managed_node_groups = {
+
+    default_nodes = {
+      name = "default"
```

Comment:
What are these nodes used for?

Reply:
For ARC, the autoscaler, proxy, and all K8S services pods...

For runners nodes, I added a taint as protection, so no pod can use them without explicit toleration.